app/services/storage.js

```diff
@@ -78,7 +84,51 @@ export class ProfileStorage {
     return max;
   }
 
-  async savePlace(url, now = microtime.now()) {
+  async init() {
+    const schema = new ProfileStorageSchemaV1();
+    const v = await schema.createOrUpdate(this);
+
+    if (v !== schema.version) {
+      throw new Error(`Incorrect version ${v}; expected ${schema.version}.`);
+    }
+
+    const loaded = await this.loadPlaces();
+    console.log(`Loaded ${loaded} places.`);
+
+    return this;
+  }
+
+  /**
+   * Begins a transaction and runs `f`. If `f` returns a promise that rejects,
+   * rolls back the transaction and passes through the rejection. If `f` resolves
+   * to a value, commits the transaction and passes through the resolved value.
+   *
+   * @param f the function to call within a transaction.
+   * @returns {*} the result or rejection of `f`.
```

Comment:
All of these comments say returns `{*}` but really they return a `Promise` that resolves with `{*}`